src/uiTest/resources/project.rhino.js

```diff
@@ -159,7 +290,59 @@ function Project(underlyingProject) {
       sleep();
       button = getButton();
     }
-    robot.click(button);
+    prettyClick(button, MouseButton.LEFT_BUTTON);
+  };
+
+  const findAndClickContextMenuAction = function (name) {
+    const actionMenuItem = getComponentByClassAndText('com.intellij.openapi.actionSystem.impl.ActionMenuItem', name);
+    prettyClick(actionMenuItem, MouseButton.LEFT_BUTTON);
+  };
+
+  this.findAndResizeIdeFrame = function () {
+    const ideFrame = getComponentByClass('com.intellij.openapi.wm.impl.IdeFrameImpl');
+    let ideFrameFixture = new FrameFixture(ideFrame);
+    let dimension = new Dimension(1024, 768);
+    ideFrameFixture.resizeTo(dimension);
+  };
+
+  const getComponentByClass = function (className) {
+    return getComponent(
+        className,
+        /* text */ null,
+        /* textCmp */ function (_, _) { return true; }
+    );
+  };
+
+  const getComponentByClassAndText = function (className, text, cmpText) {
+    if (cmpText === undefined) { // default
+      cmpText = function (t, c) { return t.equals(c.getText()); };
+    }
+    return getComponent(className, text, cmpText);
+  };
+
+  const getComponent = function (className, text, textCmp) {
+    const searchForComponent = function () {
+      const result = robot.finder().findAll(component =>
+        className.equals(component.getClass().getName()) && textCmp(text, component)
+      ).toArray();
+      return result.length === 1 ? result[0] : null;
+    }
+    // The action is invoked asynchronously, let's first make sure the component has already appeared.
+    let component = searchForComponent();
+    while (component === null) {
+      sleep();
+      component = searchForComponent();
+    }
+    return component;
+  };
+
+  const prettyClick = function (component, mouseButton) {
+    robot.moveMouse(component);
+    // wait for a while before clicking
```

Comment:
Oh? as in, that's to wait for the cursor to actually move? I.e. `moveMouse` is **a**synchronous?

Reply:
nope. It looks better - the spectator watching the gif recording has more time to see what is clicked

Reply:
Okay, then mention in the comment pls